### PR TITLE
feat: support use of Context with RequestBuilder

### LINKED ISSUE #77: Support for context timeout/deadline for requests in underlying HTTP client
**Is your feature request related to a problem? Please describe.**
The default behavior for whenever a request is made is using `http.NewRequest` [(from here)](https://github.com/IBM/go-sdk-core/blob/master/v4/core/request_builder.go#L309) which will use the default HTTP clients timeout [(from here)](https://github.com/IBM/go-sdk-core/blob/master/v4/core/base_service.go#L86). This will block the caller and force them to wait until the request either fulfills or times out.

**Describe the solution you'd like**
Support passing in a context throughout different request types, e.g.:
```go
ctx, cancelFunc := context.WithDeadline(context.Background(), someTime)
client.GetDocumentAsStream(ctx, s.NewGetDocumentOptions(someDatabaseName, someID))
```
By supporting this, the request can be canceled so that the caller isn't blocked by the library if they have their own timeout they need to support.

### PR DESCRIPTION
**Pull request by padamstx**
Fixes arf/planning-sdk-squad#2230
Fixes #77

This commit adds support for setting a Context
instance on the RequestBuilder struct.  This Context instance
will then be associated with the http.Request instance constructed
by the RequestBuilder.Build() method.
The Context can be used to define a timeout/deadline for a request
or to cancel an in-flight request.

### CODE REVIEW

**Comment by padamstx** [comment]
@jorge-ibm @mkistler This is a replacement PR for the original which I have closed.

File: v4/core/request_builder.go [hunk 1]
```diff
@@ -76,6 +77,10 @@ type RequestBuilder struct {
 	// the "Content-Encoding" header will be added to the request with the
 	// value "gzip".
 	EnableGzipCompression bool
+
+	// RequestContext is an optional Context instance to be associated with the
+	// http.Request that is constructed by the Build() method.
+	ctx context.Context
```
Comment (jorge-ibm) [inline v4/core/request_builder.go:83]: Do we want this field to be exportable/public ?

**Review by jorge-ibm — APPROVED** [review]
LGTM. Approved with 1 question

**Review by mkistler — APPROVED** [review]
Looks good! 👍

File: v4/core/request_builder.go (hunk 1 above)
Comment (padamstx) [reply v4/core/request_builder.go:83]: Nope, I wanted to intentially "hide" it behind the "WithContext()" method

**Pull request merged**